src/routing/routes.ts

```
@@ -0,0 +1,114 @@
+import { HttpMethod, HttpRequest, HttpResponse } from "../http";
+import { HttpHandler } from "../http4ts";
+import { pathToRegexp, Key } from "./path-to-regexp";
+
+export interface RoutedHttpRequest extends HttpRequest {
+  routeParams: Record<string, string>;
+}
+
+export type RoutedHttpHandler = (
+  req: RoutedHttpRequest
+) => HttpResponse | Promise<HttpResponse>;
+
+export type UriTemplate = string;
+
+interface RouteDefinition {
+  method?: HttpMethod;
+  uriTemplate?: RegExp;
+  keys?: Key[];
+  handler: RoutedHttpHandler;
+}
+
+function defaultNotFoundHandler(): HttpResponse {
+  return {
+    body: "Not Found",
+    headers: {},
+    status: 404
+  };
+}
+export function routes(...routeDefinitions: RouteDefinition[]): HttpHandler {
+  let notFoundHandler: RoutedHttpHandler = defaultNotFoundHandler;
+  const lastDefinition = routeDefinitions[routeDefinitions.length - 1];
```

Should we throw an error when no route definition is provided ?